src/pages/api/v1/youfocus/index.ts

```diff
@@ -0,0 +1,205 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { connectDB } from '@/middlewares';
+import { apiStatusCodes } from '@/constant';
+import { Playlist } from '@/database';
+import { Video } from '@/interfaces';
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  await connectDB();
+  const { method } = req;
+
+  switch (method) {
+    case 'POST':
+      return handleAddPlaylist(req, res);
+    case 'GET':
+      return handleGetAllPlaylists(req, res);
+    default:
+      return res.status(apiStatusCodes.BAD_REQUEST).json({
+        success: false,
+        message: `Method ${method} not allowed`,
+      });
+  }
+};
+
+// Function to handle adding a new playlist
+const handleAddPlaylist = async (req: NextApiRequest, res: NextApiResponse) => {
+  try {
+    const { playlistUrl } = req.body;
+
+    if (!playlistUrl) {
+      return res.status(apiStatusCodes.BAD_REQUEST).json({
+        success: false,
+        message: 'Playlist URL is required',
+      });
+    }
+
+    // Extract playlist ID from the URL
+    const playlistId = extractPlaylistId(playlistUrl);
+    if (!playlistId) {
+      return res.status(apiStatusCodes.BAD_REQUEST).json({
+        success: false,
+        message: 'Invalid playlist URL',
+      });
+    }
+
+    // Check if the playlist ID already exists in the database
+    const existingPlaylist = await Playlist.findOne({ playlistId });
+    if (existingPlaylist) {
+      return res.status(apiStatusCodes.OKAY).json({
+        success: false,
+        message: 'Playlist already exists in the database',
+      });
+    }
```

Comment:
Add DB Related things do it in Query Folder. You already have get Playlist by ID function use that. Don't write same code multiple times.